packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -245,16 +245,19 @@ function useFocusContainment(scopeRef: RefObject<HTMLElement[]>, contain: boolea
     };
 
     let onBlur = (e) => {
-      let isInAnyScope = isElementInAnyScope(e.relatedTarget, scopes);
-
-      if (!isInAnyScope) {
-        activeScope = scopeRef;
-        focusedNode.current = e.target;
-        // Firefox doesn't shift focus back to the Dialog properly without this
-        raf.current = requestAnimationFrame(() => {
-          focusedNode.current.focus();
-        });
-      }
+      rafArray.current[0] = requestAnimationFrame(() => {
+        // Fall back to document.activeElement if e.relatedTarget is null (e.g. user clicks inside a iframe in a dialog)
+        let isInAnyScope = isElementInAnyScope(e.relatedTarget || document.activeElement, scopes);
+
+        if (!isInAnyScope) {
+          activeScope = scopeRef;
+          focusedNode.current = e.target;
+          // Firefox doesn't shift focus back to the Dialog properly without this
+          rafArray.current[1] = requestAnimationFrame(() => {
```

Comment:
Hmm, I've been doing some reading since I'm not very familiar with requestAnimationFrame but I think this works? Basically storing the two requestAnimationFrames in index 0 and 1 of a ref array (overwriting these values when onBlur is called again), and then canceling them on unmount. 

I thought about doing something like `rafArray.current.push(requestAnimationFrame((... etc` but that would track all requestAnimationFrames here and attempt to cancel all of them even though the vast majority would be done.

Reply:
Does it work with just a single requestAnimationFrame instead of two nested ones?

Reply:
ah good idea, yeah it looks like firefox still focuses the dialog on underlay click